File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/FilesActivityPresenter.java
package com.tevinjeffrey.vapor.ui.files;

import com.tevinjeffrey.vapor.ui.base.Presenter;

public interface FilesActivityPresenter extends Presenter<FilesActivityView> {
    void loadEmail();
    void setNavContext(FilesActivityPresenterImpl.NavContext navContext);
    FilesActivityPresenterImpl.NavContext getNavContext();

    enum NavContext {
        ALL("All Recent Files"), POPULAR("Popular"), FAVORITE("Favorites"), TRASH("Trash");

        private final String title;

        NavContext(String title) {
            this.title = title;
        }

        @Override
        public String toString() {
            return title;
        }
    }
}

File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/FilesActivityPresenter.kt
package com.tevinjeffrey.vapor.ui.files

import com.tevinjeffrey.vapor.ui.base.Presenter

interface FilesActivityPresenter : Presenter<FilesActivityView> {
    fun loadEmail()
    var navContext: NavContext
    fun refreshClicked()

    enum class NavContext private constructor(private val title: String) {
        ALL("All Recent Files"), POPULAR("Popular"), FAVORITE("Favorites"), TRASH("Trash");

        override fun toString(): String {
            return title
        }
    }
}